src/engine/ruleset/decoders/sophos/utm/utm.yml

```diff
@@ -0,0 +1,244 @@
+      - url.extension: +r_ext/$~tmp.url.original/(\w+)\?
+      - url.fragment: $~tmp.url.fragment
+      - url.original: $~tmp.url.original
+      - url.path: $~tmp.url.path
+      - url.query: $~tmp.url.query
+      - url.scheme: $~tmp.url.scheme
+      - user.name: $~tmp.user
+
+  - check:
+    - event.code: smtpd
+    map:
+      - rsa.internal.messageid: $event.code
+      - process.pid: $~tmp.process_pid
+      - related.host: +a_append/$host.hostname
+      - rsa.network.alias_host: +a_append/$host.hostname
+      - rsa.time.event_time: $@timestamp
+      - rsa.internal.event_desc: +r_ext/$~tmp.message/] ([^\]]+)$
+
+  - check:
+    - event.code: reverseproxy
+    map:
+      - rsa.internal.messageid: $event.code
+      - process.pid: $~tmp.process_pid
+      - rsa.network.alias_host: +a_append/$host.hostname
+      - rsa.time.event_time: $@timestamp
+      - rsa.misc.event_log: $~tmp.eventlog
+      - rsa.misc.severity: $~tmp.severity
+
+  - check:
+    - event.code: httpproxy
+    map:
+      - rsa.internal.messageid: $event.code
+      - process.pid: $~tmp.process_pid
+      - rsa.time.event_time: $@timestamp
+      - ~tmp.message: +r_ext/$message/] ([^\]]+)$
+      - rsa.internal.event_desc: +r_ext/$~tmp.message/] ([^\]]+)$
+
+  - check:
+    - event.code: afcd
+    map:
+      - rsa.internal.messageid: $event.code
+      - rsa.time.event_time: $@timestamp
+      - rsa.internal.event_desc: +r_ext/$~tmp.message/] ([^\]]+)$
+
+  - check:
+    - event.code: sshd
+    map:
+      - rsa.internal.messageid: $event.code
+      - rsa.time.event_time: $@timestamp
+      - rsa.internal.event_desc: +r_ext/$~tmp.message/] ([^\]]+)$
+      - rsa.network.alias_host: +a_append/$host.hostname
+
+# TODO following decoders
+  - check:
+    - event.code: barnyard
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: confd-sync
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: exim
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: frox
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: httpd
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: ipsec_starter
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: named
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: pluto
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: pop3proxy
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: sockd
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: ulogd
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: URID
+    map:
+      - rsa.internal.messageid: $event.code
+
+  - check:
+    - event.code: xl2tpd
+    map:
+      - rsa.internal.messageid: $event.code
```

Comment:
If you know the `event.code` value then maybe it is more performant to just assign such a value rather than referencing to it through a variable. On the other hand, are there any case that `rsa.internal.messageid` should not contain the value stored in `event.code`?